app.py

```diff
@@ -130,56 +138,68 @@ def _end_bor_chain_callback(result, start_time: float):
 def _update_snapshot_timestamp(*args) -> Dict[str, Any]:
     """
     Update the snapshot timestamp with the updated roles data.
     """
-    updated_roles = list(itertools.chain.from_iterable(args[0]))
-    snapshot_meta = {}
-    database_meta = {}
-    start_time = time.time()
-    for role in updated_roles:
-        if role:
-            # Generate snapshot meta
-            snapshot_meta.update(
-                {f"{k}.json": MetaFile(v["version"]) for k, v in role.items()}
-            )
-            # Generate database meta
-            database_meta.update(
-                {
-                    k: (v["expire"], v["version"])
-                    for k, v in role.items()
-                    if k != "targets"
-                }
-            )
-    logging.info(
-        "Time parsing _update_snapshot_timestamp: "
-        f"{time.time() - start_time} seconds"
-    )
+    try:
+        updated_roles = list(itertools.chain.from_iterable(args[0]))
+        snapshot_meta = {}
+        database_meta = {}
+        start_time = time.time()
+        for role in updated_roles:
+            if role:
+                # Generate snapshot meta
+                snapshot_meta.update(
+                    {f"{k}.json": MetaFile(v["version"]) for k, v in role.items()}
+                )
+                # Generate database meta
+                database_meta.update(
+                    {
+                        k: (v["expire"], v["version"])
+                        for k, v in role.items()
+                        if k != "targets"
+                    }
+                )
+        logging.info(
+            "Time parsing _update_snapshot_timestamp: "
+            f"{time.time() - start_time} seconds"
+        )
 
-    repository._update_timestamp(
-        repository.update_snapshot(snapshot_meta, database_meta).signed.version
-    )
-    logging.info(
-        "Time updating _update_snapshot_timestamp: "
-        f"{time.time() - start_time} seconds"
-    )
+        repository._update_timestamp(
+            repository.update_snapshot(snapshot_meta, database_meta).signed.version
+        )
+        logging.info(
+            "Time updating _update_snapshot_timestamp: "
+            f"{time.time() - start_time} seconds"
+        )
 
-    logging.info(
-        f"Updated snapshot/timestamp with {len(updated_roles)} role(s)"
-    )
+        logging.info(
+            f"Updated snapshot/timestamp with {len(updated_roles)} role(s)"
+        )
+    except SoftTimeLimitExceeded:
+        logging.warning(
+            f"Task _update_snapshot_timestamp reached soft time limit. "
+            "Cleaning up."
+        )
+        return {"status": "error", "message": "Task time limit exceeded."}
 
 
-@app.task(serializer="json", queue="rstuf_internals")
+@app.task(serializer="json", queue="rstuf_internals" , task_time_limit=TASK_TIME_LIMIT , task_soft_time_limit=TASK_SOFT_TIME_LIMIT)
```

Comment:
I'm not sure if we should put the `task_[soft]_limit` for the `bump_online_role` right now, for some reasons:
- It will conflict directly with the `BOR_LOCK` we have to handle
- Depending on the number of delegated online roles, it will be a problem
- Using the same task limit time to the add/remove artifacts tasks

I'm in favor here of filing a new specific issue for adding task time limit for `bump_online_roles`